client/ayon_core/hosts/substancepainter/plugins/publish/extract_textures.py

```diff
@@ -25,19 +25,24 @@ class ExtractTextures(publish.Extractor,
     def process(self, instance):
 
         config = instance.data["exportConfig"]
-        result = substance_painter.export.export_project_textures(config)
-
-        if result.status != substance_painter.export.ExportStatus.Success:
-            raise KnownPublishError(
-                "Failed to export texture set: {}".format(result.message)
-            )
-
-        # Log what files we generated
-        for (texture_set_name, stack_name), maps in result.textures.items():
-            # Log our texture outputs
-            self.log.info(f"Exported stack: {texture_set_name} {stack_name}")
-            for texture_map in maps:
-                self.log.info(f"Exported texture: {texture_map}")
+        creator_attrs = instance.data["creator_attributes"]
+        export_channel = creator_attrs.get("exportChannel", [])
+        node_ids = instance.data.get("selected_node_id", [])
+
+        with supsend_publish_layer_stack(node_ids, export_channel):
```

Comment:
Admittedly `suspend_publish_layer_stack` seems like an odd function name? It's unclear what it does.

```python
set_channel_opacity(node_ids, export_channel)
set_channel_export_status(node_ids, export_channel)
```

Preferably the `lib` function also isn't tied to _just_ publishing but is of a generic nature of doing something with the DCC software, which is what the function name and docstring preferably explain.

In this case I must admit I have no idea what the "node ids" are and why setting a channel `opacity` to zero makes it so that it does not export at all (seems odd?) but if that's what it does, then I'd make _that_ the function description, etc.

Reply:
> Admittedly `suspend_publish_layer_stack` seems like an odd function name? It's unclear what it does.
> 
> ```python
> set_channel_opacity(node_ids, export_channel)
> set_channel_export_status(node_ids, export_channel)
> ```
> 
> Preferably the `lib` function also isn't tied to _just_ publishing but is of a generic nature of doing something with the DCC software, which is what the function name and docstring preferably explain.
> 
> In this case I must admit I have no idea what the "node ids" are and why setting a channel `opacity` to zero makes it so that it does not export at all (seems odd?) but if that's what it does, then I'd make _that_ the function description, etc.

This exports the selected layer stack, which means the other layer stacks on top of that won't be showed up at the map when publishing.

Reply:
Interesting - but then `set_layer_stack_opacity` or whatever makes more sense since that's what it'd be doing?